*** tor-b600495/src/or/channeltls.c	2013-12-05 09:30:11.000000000 -0800
--- tor-bugfix/src/or/channeltls.c	2013-12-07 00:43:09.438040392 -0800
*************** channel_tls_process_netinfo_cell(cell_t
*** 1435,1441 ****
    my_addr_ptr = (uint8_t*) cell->payload + 6;
    end = cell->payload + CELL_PAYLOAD_SIZE;
    cp = cell->payload + 6 + my_addr_len;
!   if (cp >= end) {
      log_fn(LOG_PROTOCOL_WARN, LD_OR,
             "Addresses too long in netinfo cell; closing connection.");
      connection_or_close_for_error(chan->conn, 0);
--- 1435,1441 ----
    my_addr_ptr = (uint8_t*) cell->payload + 6;
    end = cell->payload + CELL_PAYLOAD_SIZE;
    cp = cell->payload + 6 + my_addr_len;
!   if (my_addr_len >= CELL_PAYLOAD_SIZE - 6) {
      log_fn(LOG_PROTOCOL_WARN, LD_OR,
             "Addresses too long in netinfo cell; closing connection.");
      connection_or_close_for_error(chan->conn, 0);
